src/ReverseMarkdown/Converters/Code.cs

```diff
@@ -11,7 +13,51 @@ public Code(Converter converter) : base(converter)
 
         public override string Convert(HtmlNode node)
         {
-            return $"`{System.Net.WebUtility.HtmlDecode(node.InnerText.Trim())}`";
+            // Depending on the content "surrounding" the <code> element,
+            // leading/trailing whitespace is significant. For example, the
+            // following HTML renders as expected in a browser (meaning there is
+            // proper spacing between words):
+            //
+            //   <p>The JavaScript<code> function </code>keyword...</p>
+            //
+            // However, if we simply trim the contents of the <code> element,
+            // then the Markdown becomes:
+            //
+            //   The JavaScript`function`keyword...
+            //
+            // To avoid this scenario, do *not* trim the inner text of the
+            // <code> element.
+            //
+            // For the HTML example above, the Markdown will be:
+            //
+            //   The JavaScript` function `keyword...
+            //
+            // While it might seem preferable to try to "fix" this by trimming
+            // the <code> element and insert leading/trailing spaces as
+            // necessary, things become complicated rather quickly depending
+            // on the particular content. For example, what would be the
+            // "correct" conversion of the following HTML?
+            //
+            //   <p>The JavaScript<b><code> function </code></b>keyword...</p>
+            //
+            // The simplest conversion to Markdown:
+            //
+            //   The JavaScript**` function `**keyword...
+            //
+            // Some other, arguably "better", alternatives (that would require
+            // substantially more conversion logic):
+            //
+            //   The JavaScript** `function` **keyword...
+            //
+            //   The JavaScript **`function`** keyword...
+
+            var sb = new StringBuilder();
+
+            sb.Append('`');
```

Comment:
@jeremy-jameson  isn't the fix just `return $"`{System.Net.WebUtility.HtmlDecode(node.InnerText)}`"` by removing `.Trim()`? I think we don't need using `StringBuilder` here. Anyhow, I am pulling in the changes.

Reply:
To understand the reason for introducing `StringBuilder`, take a look at the [full file for commit 5b67ef9](https://github.com/mysticmind/reversemarkdown-net/blob/5b67ef94f5d5883877a1f84cfaf8dcb5cb7aef22/src/ReverseMarkdown/Converters/Code.cs).

I originally wanted to keep the `Trim` call but add spaces before and after the enclosing backquotes for the content within the `<code>` element. However, it didn't feel right to _always_ add spaces, so I added some logic to check if the previous/next nodes are `#text` elements that already end/start with whitespace (in which case adding spaces would not only be unnecessary, but doing so would actually change the generated Markdown).

That simple approach works great when dealing with content like the following:

```HTML
The JavaScript<code> function </code>keyword...
```

However, what about content where the `<code>` element resides within another inline element:

```HTML
The JavaScript<b><code> function </code></b>keyword...
```

In this particular case, it seems like the leading/trailing spaces should be moved before/after the opening and closing `<b>` tags. In other words, some kind of logic must be performed to figure out where the spaces should be inserted.

> FYI: You can see where I actually do this in a "preprocessing" step that I run before converting one of my blog posts to Markdown:
>
> https://github.com/jeremy-jameson/BlogML2Hugo/blob/master/Core/FixHtmlFormattingIssuesStep.cs
>
> However, it is worth noting that I first perform that "fix-up" on `<code>` elements (including nested `<span>` elements) and then on various "emphasis" elements (specifically `<b>`, `<em>`, `<i>`, and `<strong>`) -- in order to get the leading/trailing whitespace into the desired location.

To avoid that additional complexity, I simplified the code to remove the previous/next whitespace check and simply removed the call to `Trim` ([Code.cs from commit bb48c16](https://github.com/mysticmind/reversemarkdown-net/blob/bb48c16be332e1a561c0887057b54cdf4fbb15f1/src/ReverseMarkdown/Converters/Code.cs)).

Reply:
Sounds fine @jeremy-jameson